File: csharp/Business/Services/RallyService.cs
using Domain.Models;
using Repository.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Business.Services
{
    public class RallyService
    {
        private IRallyRepository rallyRepository;

        public RallyService(IRallyRepository rallyRepository)
        {
            this.rallyRepository = rallyRepository;
        }

        public Rally AddRally(int engineCapacity)
        {
            Rally rally = new Rally(engineCapacity);
            return rallyRepository.Add(rally);
        }

        public ICollection<Rally> GetAllRallies()
        {
            return rallyRepository.FindAll();
        }
    }
}

File: java/Business/BusinessImpl/src/main/java/com/rallies/business/impl/RallyService.java
package com.rallies.business.impl;

import com.rallies.dataaccess.api.RallyRepository;
import com.rallies.domain.models.Rally;

import java.util.Collection;

public class RallyService {
    private RallyRepository rallyRepository;

    public RallyService(RallyRepository rallyRepository) {
        this.rallyRepository = rallyRepository;
    }

    public Rally addRally(int engineCapacity) {
        Rally rally = new Rally(engineCapacity);
        return rallyRepository.save(rally);
    }

    public Collection<? extends Rally> getAllRallies() {
        return rallyRepository.findAll();
    }
}
